### PR TITLE
Swap fio_hdfsio_init and fio_hdfsio_io_u_init

### PR DESCRIPTION
**Pull request by albert-chang0**
Commit 08dc3bd50 initialized the io engine before the io_u buffers. The
unintended consequence for the libhdfs io engine was that the HDFS
connection was not established before attempting to open HDFS file
handles. This caused a NPE in the java layer on start-up preventing the
libhdfs io engine from being able to be used.

Also fix usage of JAVA_HOME environmental variable.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -60,7 +60,7 @@ SOURCE :=	$(sort $(patsubst $(SRCDIR)/%,%,$(wildcard $(SRCDIR)/crc/*.c)) \
 
 ifdef CONFIG_LIBHDFS
   HDFSFLAGS= -I $(JAVA_HOME)/include -I $(JAVA_HOME)/include/linux -I $(FIO_LIBHDFS_INCLUDE)
-  HDFSLIB= -Wl,-rpath $(JAVA_HOME)/jre/lib/$(FIO_HDFS_CPU)/server -L$(JAVA_HOME)/jre/lib/$(FIO_HDFS_CPU)/server $(FIO_LIBHDFS_LIB)/libhdfs.a -ljvm
+  HDFSLIB= -Wl,-rpath $(JAVA_HOME)/lib/$(FIO_HDFS_CPU)/server -L$(JAVA_HOME)/lib/$(FIO_HDFS_CPU)/server $(FIO_LIBHDFS_LIB)/libhdfs.a -ljvm
```
Comment (sitsofe) [inline Makefile:63]: Could you split this fix into a seperate commit?

**Review by sitsofe — CHANGES_REQUESTED** [review]
Some superficial comments. Could you:
- Prefix the area on your commit titles (see https://github.com/axboe/fio/pull/1050#issuecomment-663606909 )? Maybe `engines/hdfs:` ?
- Add your Signed-off-by ?

File: Makefile (hunk 1 above)
Comment (albert-chang0) [reply Makefile:63]: Separate commit and PR? Or just separate commit within this PR is fine?

Comment (sitsofe) [reply Makefile:63]: Same PR just seperate commits as you've done

**Comment by sitsofe** [comment]
@axboe this one LGTM...

**Review by sitsofe — APPROVED** [review]

**Pull request merged**